app/scripts/controller-init/confirmations/transaction-controller-init.ts

```diff
@@ -0,0 +1,304 @@
+  getIsSmartTransaction,
+  getSmartTransactionsPreferenceEnabled,
+  isHardwareWallet,
+} from '../../../../shared/modules/selectors';
+import {
+  SmartTransactionHookMessenger,
+  submitSmartTransactionHook,
+} from '../../lib/transaction/smart-transactions';
+import { trace } from '../../../../shared/lib/trace';
+///: BEGIN:ONLY_INCLUDE_IF(build-mmi)
+import {
+  afterTransactionSign as afterTransactionSignMMI,
+  beforeCheckPendingTransaction as beforeCheckPendingTransactionMMI,
+  beforeTransactionPublish as beforeTransactionPublishMMI,
+  getAdditionalSignArguments as getAdditionalSignArgumentsMMI,
+} from '../../lib/transaction/mmi-hooks';
+///: END:ONLY_INCLUDE_IF
+import {
+  handlePostTransactionBalanceUpdate,
+  handleTransactionAdded,
+  handleTransactionApproved,
+  handleTransactionConfirmed,
+  handleTransactionDropped,
+  handleTransactionFailed,
+  handleTransactionRejected,
+  handleTransactionSubmitted,
+  TransactionMetricsRequest,
+} from '../../lib/transaction/metrics';
+import {
+  ControllerInitFunction,
+  ControllerInitRequest,
+  ControllerInitResult,
+} from '../types';
+import { TransactionControllerInitMessenger } from '../messengers/transaction-controller-messenger';
+import { ControllerFlatState } from '../controller-list';
+
+export const TransactionControllerInit: ControllerInitFunction<
+  TransactionController
+> = (request) => {
+  const {
+    controllerMessenger,
+    initMessenger,
+    getFlatState,
+    getGlobalChainId,
+    getPermittedAccounts,
+    getTransactionMetricsRequest,
+    persistedState,
+  } = request;
+
+  const {
+    gasFeeController,
+    keyringController,
+    networkController,
+    onboardingController,
+    preferencesController,
+    smartTransactionsController,
+    ///: BEGIN:ONLY_INCLUDE_IF(build-mmi)
+    transactionUpdateController,
+    ///: END:ONLY_INCLUDE_IF
+  } = getControllers(request);
+
+  const controller: TransactionController = new TransactionController({
+    getCurrentNetworkEIP1559Compatibility: () =>
+      // @ts-expect-error Controller type does not support undefined return value
+      initMessenger.call('NetworkController:getEIP1559Compatibility'),
+    getCurrentAccountEIP1559Compatibility: async () => true,
+    // @ts-expect-error Mismatched types
+    getExternalPendingTransactions: (address) =>
+      getExternalPendingTransactions(smartTransactionsController(), address),
+    getGasFeeEstimates: (...args) =>
+      gasFeeController().fetchGasFeeEstimates(...args),
+    getNetworkClientRegistry: (...args) =>
+      networkController().getNetworkClientRegistry(...args),
+    getNetworkState: () => networkController().state,
+    // @ts-expect-error Controller type does not support undefined return value
+    getPermittedAccounts,
+    // @ts-expect-error Preferences controller uses Record rather than specific type
+    getSavedGasFees: (chainId: Hex) =>
+      preferencesController().state.advancedGasFee[chainId],
+    incomingTransactions: {
```

Comment:
Nit: I see that `etherscanApiKeysByChainId` was removed here. Maybe best to remove that in a separate PR, so that the reason for its removal isn't lost in the git history amidst the rest of this refactor

Reply:
Reverted for clarity, though can confirm there should be zero impact.